src/libraries/System.Private.CoreLib/src/System/Reflection/FieldAccessor.cs

```diff
@@ -13,7 +13,7 @@ internal sealed class FieldAccessor
         private readonly RtFieldInfo _fieldInfo;
         private IntPtr _addressOrOffset;
         private unsafe MethodTable* _methodTable;
-        private FieldAccessorType _fieldAccessType;
+        private volatile FieldAccessorType _fieldAccessType;
```

Comment:
`_fieldAccessType` is used as the gate for lock-free initialization, so it needs to be volatile and always set last.

Reply:
> and always set last.

```
field1 = value1;
volatile_field = value2;
```
we don't need to mark field1 as volatile too to guarantee that?

Reply:
My understanding of the volatile semantic tells me it cannot, just wanted to double check. e.g. on arm64 it's going to be:
```
str reg1, [<addr + cns1>]
stlr reg2, [<addr + cns2>] ;; volatile
```

Reply:
Right, we have it written down in https://github.com/dotnet/runtime/blob/main/docs/design/specs/Memory-model.md#order-of-memory-operations